py/server/deephaven/_wrapper.py

```diff
@@ -35,6 +35,34 @@ def __init_subclass__(cls, *args, **kwargs):
     def j_object(self) -> jpy.JType:
         ...
 
+    def __repr__(self):
+        self_type = type(self)
+        return f"{self_type.__module__}.{self_type.__qualname__}({repr(self.j_object)})"
+
```

Comment:
I think we want to listen to what @chipkent will say about it.

Reply:
So, there is a big difference I think in what the community does vs what the "standard" says.

https://docs.python.org/3/reference/datamodel.html#object.__repr__

>  If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value

https://docs.python.org/3/reference/datamodel.html#object.__str__

```python
>>> df = pd.DataFrame(
    {
        "Name": [
            "Braund, Mr. Owen Harris",
            "Allen, Mr. William Henry",
            "Bonnell, Miss. Elizabeth",
        ],
        "Age": [22, 35, 58],
        "Sex": ["male", "male", "female"],
    }
)
>>> df
                       Name  Age     Sex
0   Braund, Mr. Owen Harris   22    male
1  Allen, Mr. William Henry   35    male
2  Bonnell, Miss. Elizabeth   58  female
```

I *wish* there was a python standard method that was specifically targeted towards interactive environments (something like `__display__()`); b/c the output of a dataframe is *not* a valid Python expression.

I think having JObjectWrapper.__repr__ be like I've written it above, or similar, is pretty useful - and if we want/need to override __repr__ for "display" purposes, I think we should be mindful and do it on a class-by-class basis.

Reply:
There is also; in the DH UI, we don't print out the results of an expression. So, simply executing `<variableName>` doesn't print out anything. Maybe something we should change.

Reply:
I think there are a few things to comment on here.

1) I think this StackOverflow discussion summarizes how people treat `__str__` vs `__repr__` in the wild.  Basically, `__repr__` is used as a (debugging) tool for devs, and `__str__` is used for customers and about everything else.  As such, I think the changes here make sense with common usage and are consistent with my original comments on the python effort before @jmao-denver convinced me otherwise.
https://stackoverflow.com/questions/1436703/what-is-the-difference-between-str-and-repr

2) This quick code example shows that common use cases, such as print and f-strings default to `__str__`, as we would want.
![image](https://user-images.githubusercontent.com/5250374/164313521-a4554e0b-2bcb-439d-9592-0fb6d6100d46.png)

3) Shortly, we will have some new use cases to consider.  As we get dh-as-a-library (DaaL), we will have the (1) DH IDE, (2) python command line, and (3) Jupyter.  The DH IDE is special in that it does not print values out -- we should consider making it compliant with the other environments.  Both the python command line and jupyter print values out. ... unfortunately, both the python command line, and jupyter use `__repr__` when displaying a variable.  
![image](https://user-images.githubusercontent.com/5250374/164315268-67bfd446-6bf2-40bd-80c8-6d623020d194.png)
![image](https://user-images.githubusercontent.com/5250374/164315299-f76a4455-7c0d-43fb-9fee-3cb9e770ae85.png)

Pandas clearly uses `__repr__ == __str__`:
![image](https://user-images.githubusercontent.com/5250374/164317524-e4fac1c6-5997-42d2-91dd-ebf64d17e9e9.png)


My initial reaction is that item 3 with this change likely results in our product being less user-friendly than something like Pandas.  Unless there is a way around item 3, I think we are forced to just implement `__repr__`, as it is done before this PR.  I am ok if there is a separate method to return what the proposed `__repr__` does.  Alternatively, this could be considered on a class-by-class basis.

Reply:
This looks like the only item that was not really addressed in discussions.  @jmao-denver @devinrsmith

Reply:
I've changed the __repr__ / __str__ for objects that previously had __repr__ defined so the behavior stays the same as it is today. I think we'll want to re-evaluate this in the future.